publication/src/main/scala/hmda/publication/reports/util/ReportUtil.scala

```diff
@@ -0,0 +1,84 @@
+package hmda.publication.reports.util
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import hmda.census.model._
+import hmda.model.publication.reports.ApplicantIncomeEnum._
+import hmda.model.publication.reports._
+import hmda.publication.reports.{ AS, EC, MAT }
+import hmda.publication.reports.util.DispositionType._
+import hmda.publication.reports.util.RaceUtil.raceBorrowerCharacteristic
+import hmda.publication.reports.util.EthnicityUtil.ethnicityBorrowerCharacteristic
+import hmda.publication.reports.util.MinorityStatusUtil.minorityStatusBorrowerCharacteristic
+import hmda.query.model.filing.LoanApplicationRegisterQuery
+import hmda.util.SourceUtils
+
+import scala.concurrent.Future
+
+object ReportUtil extends SourceUtils {
+
+  def msaReport(fipsCode: String): MSAReport = {
+    val cbsa = CbsaLookup.values.find(x => x.cbsa == fipsCode).getOrElse(Cbsa())
+    val stateFips = cbsa.key.substring(0, 2)
+    val state = StateAbrvLookup.values.find(s => s.state == stateFips).getOrElse(StateAbrv("", "", ""))
+    MSAReport(fipsCode, CbsaLookup.nameFor(fipsCode), state.stateAbrv, state.stateName)
+  }
+
+  def calculateMedianIncomeIntervals(fipsCode: Int): Array[Double] = {
+    val msaIncome = MsaIncomeLookup.values.find(msa => msa.fips == fipsCode).getOrElse(MsaIncome())
+    val medianIncome = msaIncome.income / 1000
```

Comment:
Precision isn't necessary here because it's always being compared to the applicant's income, which is also an `Int`.